recipes-kernel/accton-csp7551-cpld-mod/accton-csp7551-cpld-mod.bb

```diff
@@ -0,0 +1,15 @@
+SUMMARY = "CSP 7551 cpld module"
```

Comment:
If you have a publicly available source for them I would suggest directly checking them out from there instead of creating a local copy of them; makes updating them to a new version easier.

Having all modules built from one package is fine, they don't need to be separated.

Also having them in one package avoids the need for `KBUILD_MODPOST_WARN=1`, as modpost will be then able to resolve symbols between modules and set proper dependencies, and won't error out anymore.

Reply:
The publicly available source does not look robust enough to pull from. I want to keep the source in this repo.

I agree that all modules could be built from one package, but there's no harm in having separate packages, and that's what we have done in meta-open-network-linux until we switched to building via ONL (which does not apply in this case because it is not an ONL platform).

Reply:
How about https://github.com/WeichenChenACCTON/sonic-buildimage-wedge100bf-32qs/commit/bcf779f1798fbecc6024a05cbf4e994c820a67e2 ?

This one seems pretty stable, unmodified since august 2022, identical to the source you are using (at least the sonic-platform-modules-accton/ content is identical, including the modules), and is referenced in a [somewhat stale PR to sonic-buildimage](https://github.com/sonic-net/sonic-buildimage/pull/13099), which is where I found it.

The only difference to the source you are using is the sonic-buildimage commit it is based on, which does not matter for us at all.

And while we had modules as single packages in the past doesn't mean this is the right way to do it. Especially since it breaks inter-module dependencies.

So I would ask to at least merge them into a single package (e.g. accton-csp7551-modules or so), and preferably check out from git. Makes the source immediately obvious.

Reply:
Move all modules into a single package.